### PR TITLE
Span tweaks

### PR DESCRIPTION
**Pull request by nnethercote**
Some minor improvements to code clarity.

r? @cjgillot

### CODE REVIEW

**Comment by nnethercote** [comment]
There are a couple of `njn` comments for things I'm uncertain about. In particular, the `parent` field seems like it isn't used unless `-Zincremental-relative-spans` is used, is that right? Is there any prospect of that being made the default? It's a fair amount of extra complication handling parents if it's never used in practice.

File: compiler/rustc_span/src/span_encoding.rs [hunk 1]
```diff
@@ -13,80 +7,93 @@ use rustc_data_structures::fx::FxIndexSet;
...
+///   needed for `ctxt` values depend partly on the crate size and partly on
+///   the form of the code. No crates in `rustc-perf` need more than 15 bits
+///   for `ctxt_or_parent_or_marker`, but larger crates might need more than 16
+///   bits. njn: what about parent?
 ///
+/// njn: this comment is nonsensical, what is it supposed to say?
 /// In order to reliably use parented spans in incremental compilation,
 /// the dependency to the parent definition's span. This is performed
 /// using the callback `SPAN_TRACK` to access the query engine.
 ///
```
Comment (cjgillot) [inline compiler/rustc_span/src/span_encoding.rs:77]: in relative span mode, we hide part of the information contained in the span from the query system. The `Span` actually holds `lo`, `hi` and `ctxt`, but the query system thinks it only holds `offset from parent's span lo`, `hi - lo` and `ctxt`. This file is responsible for the encapsulation. The conversion is meant to happen in `data_untracked` (untracked by the query system), and `data` which tracks it for the query system by calling `SPAN_TRACK`. No code should access `lo` or `hi` other that through one of these methods.

**Comment by nnethercote** [comment]
@cjgillot: I have updated to remove the `njn` comments. This is now ready for review. Thanks!

**Pull request merged**